Kernel/Devices/HID/PS2KeyboardDevice.cpp

```diff
@@ -107,6 +107,7 @@ UNMAP_AFTER_INIT PS2KeyboardDevice::PS2KeyboardDevice(const I8042Controller& ps2
     : IRQHandler(IRQ_KEYBOARD)
     , KeyboardDevice()
     , I8042Device(ps2_controller)
+    , m_io_work_queue(adopt_own_if_nonnull(new WorkQueue("PS2KeyboardDevice WorkQueue")).release_nonnull())
```

Comment:
Can we avoid allocating in constructors and use Factory's instead? 
I know it happens on startup and will never realistically OOM, but it establishes a bad pattern that people seem to like to copy and paste to non startup code 😢 .

Reply:
there is the `make<T>` for NonnullOwnptrs
(it is essentially the same)

Reply:
> there is the `make<T>` for NonnullOwnptrs
> (it is essentially the same)

The issue isn't the way the WorkQueue is allocated (`make<T>(..)` vs <`adopt_own_if_nonnull(..)`) ... I'm advocating for allocations to happen outside of constructors so failures can always be observed.